File: apA_analyticSolutions/matlab/cylinder.m
clear
close all

T = 1;
H = 2;
ntest = 100; 

alpha_0 = fzero(@(z) besselj(0, z), 3);
% besselj(0, alpha_0); % Should be small

fun = @(r,z)(besselj(0,(alpha_0/T)*r).*sin((pi/H)*z));

dr = T/ntest;
dz = H/ntest;
r = 0.5*dr:dr:T;
z = 0.5*dz:dz:H;

[R,Z] = meshgrid(r,z);

val = fun(R,Z)';

figure('pos',[1000 1000 650 1000]);
imagesc(r,z,val');
view([0,90]);
colormap(viridis);
colorbar;
xlabel('r [cm]');
ylabel('z [cm]');
xlim([0 T]);
ylim([0 H]);
title('Slice of Finite Cylinder');

set(gca,'FontName','Times New Roman','FontSize',20);
print('../figs/finite_cyl.eps','-depsc2');
close(gcf)

File: apA_analyticSolutions/python/cylinder.py
import numpy as np
import matplotlib.pyplot as plt
import scipy
import scipy.special

LW = 2
FN = "Times New Roman"
FS = 12
plt.rc("lines", lw=LW)
plt.rc("mathtext", fontset="stix")
plt.rc("font", family=FN, size=FS)

T = 50.0
H = 100.0
ntest = 100

alpha_0 = scipy.special.jn_zeros(0, 1)[0]
fun = lambda r, z: scipy.special.j0(alpha_0 / T * r) * np.sin(np.pi / H * z)

dr = T / ntest
dz = H / ntest

r = np.arange(0.5 * dr, T, dr)
z = np.arange(0.5 * dz, H, dz)

R, Z = np.meshgrid(r, z)
val = fun(R, Z)

plt.figure()
plt.imshow(val, extent=[0, T, 0, H])
plt.xlim((0, T))
plt.ylim((0, H))
plt.colorbar()
plt.xlabel("r [cm]")
plt.ylabel("z [cm]")
plt.title("Slice of Finite Cylinder")
plt.savefig("../figs/finite_cyl.pdf")

plt.show()
